lib/vrl/compiler/src/value/convert.rs

```diff
@@ -20,7 +20,7 @@ impl Value {
     ///
     /// This is a non-public function because we want to avoid exposing internal
     /// details about the expression variants.
```

Comment:
See this comment. At the very least, we have to remove this paragraph. But I'd like to push us towards not making this public at all.